### PR TITLE
Move dashboard authentication to users

### PR DESCRIPTION
**Pull request by govindkrjoshi**
- Creates master users for admins
- Uses master_users for authenticating dashboard users

### CODE REVIEW

File: app/controllers/email_authentications/invitations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def create
+    authorize :invitation, :create?
+    @role = params.require(:email_authentication).require(:role).downcase.to_sym
+    user = nil
+    User.transaction do
+      super do |resource|
+
+        # Temporary to make tests work
```
Comment (vkrmis) [inline app/controllers/email_authentications/invitations_controller.rb]: Need to fix this?

File: app/models/user.rb [hunk 2]
```diff
@@ -38,10 +57,33 @@ class User < ApplicationRecord
...
+
+  def self.invite!(options = {})
+    transaction do
+      now = Time.now
+      User.create(
+        options
+          .slice(:full_name, :role)
+          .merge(device_created_at: now,
+                 device_updated_at: now,
+                 sync_approval_status: sync_approval_statuses[:denied]))
```
Comment (vkrmis) [inline app/models/user.rb]: Should the default sync approval status be denied?

File: app/views/analytics/districts/show.html.erb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div class="dashboard">
   <nav class="breadcrumb">
-    <%= link_to 'All districts', admin_root_path %>
+    <%= link_to 'All districts', email_authentication_root_path %>
```
Comment (vkrmis) [inline app/views/analytics/districts/show.html.erb]: `email_authentication_root_path` isn't a great name for `organizations#index`, can we call this something else?

File: app/views/analytics/facilities/show.html.erb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="dashboard">
   <nav class="breadcrumb">
-    <%= link_to 'All districts', admin_root_path %>
+    <%= link_to 'All districts', email_authentication_root_path %>
```
Comment (vkrmis) [inline app/views/analytics/facilities/show.html.erb]: same as previous

File: lib/tasks/data_fixes.rake [hunk 5]
```diff
@@ -1,4 +1,4 @@
-require 'tasks/data_clean_up/move_user_recorded_data_to_registration_facility'
+require 'tasks/scripts/move_user_recorded_data_to_registration_facility'
```
Comment (vkrmis) [inline lib/tasks/data_fixes.rake]: Does this need to be in this PR?

File: lib/tasks/scripts/move_user_recorded_data_to_registration_facility.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
+class MoveUserRecordedDataToRegistrationFacility
```
Comment (vkrmis) [inline lib/tasks/scripts/move_user_recorded_data_to_registration_facility.rb]: Does this need to be in this PR?

File: spec/features/admins_spec.rb [hunk 7]
```diff
@@ -22,13 +22,13 @@
     end
   end
 
-  describe "editing admins" do
+  xdescribe "editing email_authentications" do
```
Comment (vkrmis) [inline spec/features/admins_spec.rb]: Uncomment and fix these specs?

File: spec/lib/tasks/data_clean_up/move_user_recorded_data_to_registration_facility_spec.rb [hunk 8]
```diff
@@ -1,5 +1,5 @@
 require 'rails_helper'
-require 'tasks/data_clean_up/move_user_recorded_data_to_registration_facility'
+require 'tasks/scripts/move_user_recorded_data_to_registration_facility'
```
Comment (vkrmis) [inline spec/lib/tasks/data_clean_up/move_user_recorded_data_to_registration_facility_spec.rb]: Does this need to be in this PR?

File: spec/mailers/previews/anonymized_data_download_mailer_preview.rb [hunk 9]
```diff
@@ -2,7 +2,7 @@
 class AnonymizedDataDownloadMailerPreview < ActionMailer::Preview
   def mail_anonymized_data
     recipient_name = 'test_admin'
-    recipient_email = 'test_admin@admins.org'
+    recipient_email = 'test_admin@email_authentications.org'
```
Comment (vkrmis) [inline spec/mailers/previews/anonymized_data_download_mailer_preview.rb]:
```suggestion
    recipient_email = 'test_admin@admins.org'
```

File: app/models/user.rb (hunk 2 above)
Comment (govindkrjoshi) [reply app/models/user.rb]: Yes, the admins don't require syncing.

File: app/views/analytics/districts/show.html.erb (hunk 3 above)
Comment (govindkrjoshi) [reply app/views/analytics/districts/show.html.erb]: This is generated by devise.

File: lib/tasks/data_fixes.rake (hunk 5 above)
Comment (govindkrjoshi) [reply lib/tasks/data_fixes.rake]: Don't want to have multiple directories with just one file in them.

**Pull request merged**